app/controllers/floors_controller.rb

```diff
@@ -54,77 +49,37 @@ def floor_params
 
   private
 
-  def fetch_persistent_locations
-    Location.where(:persistent => true)
-  end
-
   def process_search(search_params)
-    @search_results = []
-    @search_results << Location.search_for(search_params)
-    @search_results << Tag.search_for(search_params)
-    @search_results << Trait.search_for(search_params)
-    @search_results
-  end
-
-  def normalize_search_result_floors(floors)
-    @normalized_floors = [nil, nil, nil, nil, nil, nil]
-    floors.each do |floor|
-      @normalized_floors[floor.level - 1] = floor
-    end
-    @normalized_floors
-  end
-
-  def extract_maps_from_floors(floors)
-    @maps = []
-    floors.each do |floor|
-      @maps << floor.map
-    end
-    @maps
+    return [] if search_params.nil?
+    results = []
+    results << Location.search_for(search_params)
+    results << Tag.search_for(search_params)
+    results << Trait.search_for(search_params)
+    results.flatten
   end
 
   def extract_floors(locations)
-    @search_result_floors = []
-      locations.each do |location|
-        if (location.is_a?(Location))
-          @search_result_floors << location.floor unless @search_result_floors.include?(location.floor)
-        elsif(locations.first.is_a?(Tag))
-          @search_result_floors << tag.location.floor unless @search_result_floors.include?(tag.location.floor)
-        else
-
-        end
-      end
-    normalize_search_result_floors(@search_result_floors)
-  end
-
-  def extract_icon_images
-    @icon_images = {}
-    Icon.all.each do |icon|
-      @icon_images[icon.id] = icon.icon_image
+    floors = Array.new(6)
+    locations.to_a.each do |location|
```

Comment:
Im concerned about this refactor because sometimes location can actually include a tag. You would need to grab the location that is associated with that tag and then extract its floor.

Reply:
Edit. I see now that in extract locations, you are covering that, which in turn simplifies this code.